spectral_cube/spectral_cube.py

```diff
@@ -2328,6 +2328,186 @@ def _new_cube_with(self, **kwargs):
 
     _new_cube_with.__doc__ = BaseSpectralCube._new_cube_with.__doc__
 
+    def spatial_smooth_median(self, ksize,
+                        #numcores=None,
+                        **kwargs):
+        """
+        Smooth the cube along the spectral dimension
+
+        Parameters
+        ----------
+        kernel : `~astropy.convolution.Kernel1D`
+            A 1D kernel from astropy
```

Comment:
Spatial smooth should use a 2D kernel, no?

Reply:
In the median case, there is no kernel, so this should be removed.

Reply:
fixed and push coming shortly